authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/permission/ScopePolicyProviderFactory.java

```diff
@@ -62,6 +66,33 @@ public ScopePermissionRepresentation toRepresentation(Policy policy, Authorizati
         return new ScopePermissionRepresentation();
     }
 
+    @Override
+    public void onCreate(Policy policy, ScopePermissionRepresentation representation, AuthorizationProvider authorization) {
+        updateResourceType(policy, representation);
+    }
+
+    @Override
+    public void onUpdate(Policy policy, ScopePermissionRepresentation representation, AuthorizationProvider authorization) {
+        updateResourceType(policy, representation);
+    }
+
+    private void updateResourceType(Policy policy, ScopePermissionRepresentation representation) {
+        if (representation != null) {
+            //TODO: remove this check once we migrate to new API
+            if (ScopePermissionRepresentation.class.equals(representation.getClass())) {
```

Comment:
Is this `if` statement really necessary ?

Reply:
We duplicated this `if` from:
https://github.com/keycloak/keycloak/blob/8436a880755a7166741d572b7a803e0417fdb909/authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/permission/ResourcePolicyProviderFactory.java#L83
Since the API now specifies the representation type (`ResourcePermissionRepresentation` and  `ScopePermissionRepresentation`) maybe we could remove the `if` and the `class.cast` from both `ResourcePolicyProviderFactory` and `ScopePolicyProviderFactory`. Do you agree with this change?

Reply:
Yes, I agree.

Reply:
Done!